CMake/BaseCode.cmake

```diff
@@ -64,7 +64,7 @@ function(ttk_add_base_template_library library)
 
   string(TOUPPER "${CMAKE_BUILD_TYPE}" uppercase_CMAKE_BUILD_TYPE)
   if(uppercase_CMAKE_BUILD_TYPE MATCHES RELEASE)
```

Comment:
I think this whole if can be removed if this flag is already set by the new `CompilerFlags.cmake`

Reply:
(Having this flag INTERFACE was not good anyway)

Reply:
Isn't this block still useful when building an application against the base layer only?